default/python/AI/ShipDesignAI.py

```diff
@@ -1041,7 +1041,7 @@ def parse_tokens(tokendict, is_hull=False):
         if self.design_stats.fuel < 0:
             self.design_stats.fuel = 0
         else:
-            self.design_stats.fuel *= self._hull_fuel_efficiency()
+            self.design_stats.fuel = (self.design_stats.fuel - self.hull.fuel) * self._hull_fuel_efficiency() + self.hull.fuel
```

Comment:
Not introduced in this PR, so I could also fix this separately after this is merged... There is another missing fuel effect here:
```
if self.design_stats.fuel < 1: 
    self.design_stats.fuel = 0
```